includes/Block/Web_Stories_Block.php

```diff
@@ -87,10 +80,25 @@ public function register() {
 			$this->get_script_settings()
 		);
 
+		$base_path = $this->assets->get_base_path( 'assets/js/block.json' );
+		$this->register_block_type( $base_path );
+	}
+
+	/**
+	 * Registers a block type from metadata stored in the `block.json` file.
+	 *
+	 * @since 1.9.0
+	 *
+	 * @param string $base_path Path to the JSON file with metadata definition for
+	 *     the block or path to the folder where the `block.json` file is located.
+	 *
+	 * @return void
+	 */
+	protected function register_block_type( $base_path ) {
 		// Note: does not use 'script' and 'style' args, and instead uses 'render_callback'
 		// to enqueue these assets only when needed.
-		register_block_type(
-			self::BLOCK_NAME,
+		register_block_type_from_metadata(
+			$base_path,
 			[
```

Comment:
Is this second parameter still needed? All the attributes could theoretically be defined in `block.json` alone, right?

Except probably for the `'default' => __( 'Web Story', 'web-stories' )` bit as it would not be translatable in `block.json`. That said, that default could perhaps be enforced some other way.

Reply:
Yeah, the reason is those translatable default values. We could do that some other way, I am not sure the point of it.

Reply:
If we'd do those some other way, then block.json would _really_ be the single source of truth. Hence the question.

Reply:
I am not sure how we would do that.